doc/releases/changelog-dev.md

```diff
@@ -9,6 +9,9 @@
 
 <h3>Improvements 🛠</h3>
 
+* `default.clifford` now supports arbitrary state-based measurements with `qml.Snapshot`.
```

Comment:
Please add your name to the list of contributors (in the ascending order of your last name).